notebooks/02_model/lightfm_deep_dive.ipynb

```diff
@@ -0,0 +1,1358 @@
+{
```

Comment:
<p>Here two comments:</p><p>1) We follow the single responsibility pattern <a href="https://github.com/Microsoft/Recommenders/wiki/Coding-Guidelines#single-responsibility" target="_blank">https://github.com/Microsoft/Recommenders/wiki/Coding-Guidelines#single-responsibility</a>, as well as explicit is better than implicit. This effictively means that we normally have one part called fit, that trains the model and another one called predict or recommend that predicts on the test set. This can be an internal method, or if the library already has a fit, use it directly.</p><p>2) The plot shows a high overfitting, are you sure that this <code>model.fit_partial(interactions=train_interactions, epochs=1</code>  works right?</p><p>3) for consistency it would be good to use our own precision_at_k and other metrics. One example that would be very similar to your is <a href="https://github.com/microsoft/recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb" target="_blank">https://github.com/microsoft/recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb</a>. There we are using a similar approach, using an external library, called surprise, and try to add as less code as needed to show how it works</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1096/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Reply:
I've added explanation why fit_partial was used, also added example re how to use fit method and compared both models

Reply:
I have changed and restructured the notebook, now it contains steps to map the data into a format suitable for the repo's evaluation metrics.